### ISSUE TITLE
Binary links HTTP/404

### ISSUE THREAD

**Issue by huan** [frida/cryptoshark#6]
http://build.frida.re/frida/windows/Win32-Release/bin/cryptoshark-0.1.3.exe has not existed anymore.

To build the binary from scratch requires lots of work, so if anyone could share the built binary will be very apprecated, and if we can publish the binary to the Github release that would help the user a lot.

**Comment by einsteinx2** [comment]
Not only are the binary links broken but all build.frida.re links... So I can't compile it either as I can't download the QT project...

**Comment by huan** [comment]
I would like to contribute by setting a GitHub Action to build all binaries and release them as artifacts on GitHub Release because this tool is really great for debugging.

Is there any documentation about how to compile them, or I can find any build scripts anywhere? @oleavr

**Comment by oleavr** [comment]
@huan That would be awesome!  We don't have any docs on this yet, but basically on macOS you'd build it like this:

```
$ git clone --recurse-submodules https://github.com/frida/frida.git
$ cd frida
$ make core-macos-thin
$ export FRIDA=$(pwd)
$ cd frida-qml
$ qmake
$ make
$ make install
```

Then you can open this project with Qt Creator, build and run it.

**Comment by einsteinx2** [comment]
@oleavr Just tried those steps, but it failed to compile when I ran the first make command:

```
ben-mbp-wifi :: ~/workspace_frida » git clone --recurse-submodules https://github.com/frida/frida.git
Cloning into 'frida'...
remote: Enumerating objects: 8, done.
remote: Counting objects: 100% (8/8), done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 8368 (delta 3), reused 7 (delta 2), pack-reused 8360
Receiving objects: 100% (8368/8368), 13.84 MiB | 9.09 MiB/s, done.
Resolving deltas: 100% (5112/5112), done.
Submodule 'capstone' (git://github.com/frida/capstone.git) registered for path 'capstone'
Submodule 'frida-clr' (git://github.com/frida/frida-clr.git) registered for path 'frida-clr'
Submodule 'frida-core' (git://github.com/frida/frida-core.git) registered for path 'frida-core'
Submodule 'frida-gum' (git://github.com/frida/frida-gum.git) registered for path 'frida-gum'
Submodule 'frida-node' (git://github.com/frida/frida-node.git) registered for path 'frida-node'
Submodule 'frida-python' (git://github.com/frida/frida-python.git) registered for path 'frida-python'
Submodule 'frida-qml' (git://github.com/frida/frida-qml.git) registered for path 'frida-qml'
Submodule 'frida-swift' (git://github.com/frida/frida-swift.git) registered for path 'frida-swift'
Submodule 'frida-tools' (git://github.com/frida/frida-tools.git) registered for path 'frida-tools'
Submodule 'releng/meson' (git://github.com/frida/meson.git) registered for path 'releng/meson'
Cloning into '/Users/bbaron/workspace_frida/frida/capstone'...
remote: Enumerating objects: 27903, done.        
remote: Total 27903 (delta 0), reused 0 (delta 0), pack-reused 27903        
Receiving objects: 100% (27903/27903), 37.19 MiB | 5.98 MiB/s, done.
Resolving deltas: 100% (20182/20182), done.
Cloning into '/Users/bbaron/workspace_frida/frida/frida-clr'...
remote: Enumerating objects: 198, done.        
remote: Total 198 (delta 0), reused 0 (delta 0), pack-reused 198        
Receiving objects: 100% (198/198), 52.96 KiB | 398.00 KiB/s, done.
Resolving deltas: 100% (122/122), done.
Cloning into '/Users/bbaron/workspace_frida/frida/frida-core'...
remote: Enumerating objects: 17, done.        
remote: Counting objects: 100% (17/17), done.        
remote: Compressing objects: 100% (13/13), done.        
remote: Total 17305 (delta 4), reused 8 (delta 4), pack-reused 17288        
Receiving objects: 100% (17305/17305), 5.43 MiB | 6.91 MiB/s, done.
Resolving deltas: 100% (13022/13022), done.
Cloning into '/Users/bbaron/workspace_frida/frida/frida-gum'...
remote: Enumerating objects: 275, done.        
remote: Counting objects: 100% (275/275), done.        
remote: Compressing objects: 100% (142/142), done.        
remote: Total 33357 (delta 160), reused 207 (delta 133), pack-reused 33082        
Receiving objects: 100% (33357/33357), 28.08 MiB | 7.14 MiB/s, done.
Resolving deltas: 100% (24706/24706), done.
Cloning into '/Users/bbaron/workspace_frida/frida/frida-node'...
remote: Enumerating objects: 144, done.        
remote: Counting objects: 100% (144/144), done.        
remote: Compressing objects: 100% (103/103), done.        
remote: Total 1715 (delta 82), reused 95 (delta 41), pack-reused 1571        
Receiving objects: 100% (1715/1715), 497.84 KiB | 1.65 MiB/s, done.
Resolving deltas: 100% (1174/1174), done.
Cloning into '/Users/bbaron/workspace_frida/frida/frida-python'...
remote: Enumerating objects: 140, done.        
remote: Counting objects: 100% (140/140), done.        
remote: Compressing objects: 100% (100/100), done.        
remote: Total 2678 (delta 65), reused 108 (delta 40), pack-reused 2538        
Receiving objects: 100% (2678/2678), 812.83 KiB | 2.02 MiB/s, done.
Resolving deltas: 100% (1857/1857), done.
Cloning into '/Users/bbaron/workspace_frida/frida/frida-qml'...
remote: Enumerating objects: 3, done.        
remote: Counting objects: 100% (3/3), done.        
remote: Compressing objects: 100% (3/3), done.        
remote: Total 328 (delta 0), reused 1 (delta 0), pack-reused 325        
Receiving objects: 100% (328/328), 89.04 KiB | 607.00 KiB/s, done.
Resolving deltas: 100% (221/221), done.
Cloning into '/Users/bbaron/workspace_frida/frida/frida-swift'...
remote: Enumerating objects: 11, done.        
remote: Counting objects: 100% (11/11), done.        
remote: Compressing objects: 100% (9/9), done.        
remote: Total 270 (delta 2), reused 6 (delta 2), pack-reused 259        
Receiving objects: 100% (270/270), 77.91 KiB | 917.00 KiB/s, done.
Resolving deltas: 100% (152/152), done.
Cloning into '/Users/bbaron/workspace_frida/frida/frida-tools'...
remote: Enumerating objects: 29, done.        
remote: Counting objects: 100% (29/29), done.        
remote: Compressing objects: 100% (18/18), done.        
remote: Total 494 (delta 12), reused 26 (delta 9), pack-reused 465        
Receiving objects: 100% (494/494), 154.56 KiB | 989.00 KiB/s, done.
Resolving deltas: 100% (332/332), done.
Cloning into '/Users/bbaron/workspace_frida/frida/releng/meson'...
remote: Enumerating objects: 59689, done.        
remote: Total 59689 (delta 0), reused 0 (delta 0), pack-reused 59689        
Receiving objects: 100% (59689/59689), 26.47 MiB | 6.93 MiB/s, done.
Resolving deltas: 100% (41918/41918), done.
Submodule path 'capstone': checked out '18ff84eb9bffe3e404f5558790ce7daef14b1603'
Submodule path 'frida-clr': checked out '38e1bdb9392eb333610706d6dcf146e5cb6e83b0'
Submodule path 'frida-core': checked out '16f45b31d4b16244dee7e02bee5e511a42029991'
Submodule path 'frida-gum': checked out '2c121b3a440cc928ee8ba1e70a785c2e56d8c182'
Submodule 'bindings/gumjs/tinycc' (git://github.com/frida/tinycc.git) registered for path 'frida-gum/ext/tinycc'
Cloning into '/Users/bbaron/workspace_frida/frida/frida-gum/ext/tinycc'...
remote: Enumerating objects: 261, done.        
remote: Counting objects: 100% (261/261), done.        
remote: Compressing objects: 100% (48/48), done.        
remote: Total 11882 (delta 226), reused 232 (delta 213), pack-reused 11621        
Receiving objects: 100% (11882/11882), 3.76 MiB | 5.30 MiB/s, done.
Resolving deltas: 100% (8552/8552), done.
Submodule path 'frida-gum/ext/tinycc': checked out 'dbf4486c4c2dc1808a84b960ac1a7c394687ac85'
Submodule path 'frida-node': checked out '4ca870a4ccb505a6c5e3f94e70d13b51eb8feb10'
Submodule path 'frida-python': checked out '413beb9ec8bd19298a72ccc104eb681473d730fa'
Submodule path 'frida-qml': checked out 'b5b9359a97776c4ec543c8da4f40bbdb6120f3d5'
Submodule path 'frida-swift': checked out '566c21da278c48765ba2b95371e5d7ae675c493b'
Submodule path 'frida-tools': checked out '455dd8cc6696bf499efb7bcc3cd7a9037162ae52'
Submodule path 'releng/meson': checked out '3938daa321a2e05c230757a6c840ee3d4ce9d80e'
ben-mbp-wifi :: ~/workspace_frida » cd frida
ben-mbp-wifi :: ~/workspace_frida/frida ‹git master› » make core-macos-thin
FRIDA_HOST=macos-x86_64 \
		FRIDA_ACOPTFLAGS="-Os" \
		FRIDA_ACDBGFLAGS="-g3" \
		FRIDA_ASAN=no \
		FRIDA_ENV_NAME=frida_thin \
		./releng/setup-env.sh
Downloading and deploying toolchain...
tar: -: Not found in archive
tar: Exiting with failure status due to previous errors
make[1]: *** [build/frida_thin-env-macos-x86_64.rc] Error 1
make: *** [core-macos-thin] Error 2

```

**Comment by oleavr** [comment]
@einsteinx2 [This](https://github.com/frida/frida/blob/33eb37c38e689e5b7483a5c3c4927503a092bdc6/releng/setup-env.sh#L800) is failing for some reason – networking issue perhaps? Frida fetches some prebuilt tools such as the Vala compiler, and also prebuilt dependencies a bit further down in the same script. (Reminds me we should improve this script to verify the downloads – still some rough edges left in the build system.)

**Comment by einsteinx2** [comment]
@oleavr hmm just tried again with the same error. Is there any way I can get more log output to see what dependency is failing exactly?

**Comment by oleavr** [comment]
@einsteinx2 I'd recommend adding some temporary logging to `releng/setup-env.sh` – perhaps just adding `set -x` to the start of the script will be enough to see what exactly it's trying.

**Comment by einsteinx2** [comment]
@oleavr Thanks that caught it. This is the command that was failing:

```
+ tar -C /Users/bbaron/workspace_frida/frida/build/frida_thin-toolchain-macos-x86_64 -xj -
tar: -: Not found in archive
```

Turns out it was an incompatibility with gnu-tar which I had installed from brew. Normally I never have any issues keeping gnu-tar in my path in place of Apple's tar, but it looks like this command is somehow specific to Apple's version. It would be great if the command could be rewritten in a way that supports both tar versions, but it's unlikely this will be a common issue as most people don't install gnu-tar and add it to their path like I did, so it may not be worth the work unless it's trivial to change.

**Comment by oleavr** [comment]
@einsteinx2 Thanks, nice catch! Just fixed this in cdb17fd49cb325b84b6e68791bff4b4289d61061. Don't know why I didn't just pass `-f -` instead of all that useless complexity 🤦‍♂️ – must have been low on coffee when I did that 😅

**Comment by einsteinx2** [comment]
@oleavr I've got one last question. I was now able to follow all of the steps including building in QT Creator, however I can't for the life of me find the CryptoShark application... I see all of the libraries and tools built, and I can load the project in QT Creator after following your instructions above: https://github.com/frida/cryptoshark/issues/6#issuecomment-647830302

However, it doesn't seem to actually build the application... I have zero experience with with QT Creator, so I assume I'm missing something really obvious. Other than opening the project and clicking Build in QT Creator, is there anything else I need to configure or do to get the application? Thanks.

**Comment by oleavr** [comment]
@einsteinx2 Did you open the .pro file with Qt Creator? It should then prompt you to select which toolchain(s) to use – make sure you picked one when installing Qt as I seem to recall having to tick that checkbox manually – (IIRC) for some reason the installer didn't pre-select the toolchain needed for the local system.

Once opened there should be a green “play”-button in the lower left corner.

**Comment by oleavr** [comment]
Just released CryptoShark 0.2.0 w/ macOS .dmg:
https://github.com/frida/cryptoshark/releases/tag/0.2.0

Should hopefully be easy to build for other OSes too. If any of you would be interested in helping out with the CI please do let me know.

Cheers!
